src/util/cmdlineargs.h

```diff
@@ -58,28 +60,37 @@ class CmdlineArgs {
     bool getMidiDebug() const { return m_midiDebug; }
     bool getDeveloper() const { return m_developer; }
     bool getSafeMode() const { return m_safeMode; }
+    bool getSettingsPathSet() const { return m_settingsPathSet; }
     bool getTimelineEnabled() const { return !m_timelinePath.isEmpty(); }
     const QString& getLocale() const { return m_locale; }
     const QString& getSettingsPath() const { return m_settingsPath; }
+    void setSettingsPath(QString newSettingsPath) { m_settingsPath=QString(newSettingsPath); return; }
```

Comment:
minor note: 

```c++
m_settingsPath = newSettingsPath; 
```
Is all that's needed -- return is implicit for a void function and the QString(..) creates a temporary QString which is then assigned to m_settingsPath.